### PR TITLE
Add Pay as Business option for invoices sent to an individual who is admin of their primary workspace

### PR DESCRIPTION
**Pull request by VickyStash**
### Details

This PR implements the Pay as business option in the App for the individuals who are admins of their primary workspace.

### Fixed Issues

$ https://github.com/Expensify/App/issues/40438
PROPOSAL: N/A

### Tests

- [x] Verify that no errors appear in the JS console

1. Login to the app as User A. The user should be the admin of any workspace.
2. Go to FAB -> Select "Send invoice"
3. Send Invoice to the User B (user should be the admin of his primary workspace)
5. Login as User B and open the invoice room from step 3.
6. Tap Pay button and select `Pay as business` -> `Pay elsewhere`
7. Verify that:
- the invoice got paid
-  the room is turned into B2B invoices room, the title and avatars got updated.
8. Login as User A and open the same invoice room. Verify B2B invoices room looks as expected for the invoice sender as well.

### Offline tests

Repeat steps 5-7 from the **Tests** section.

### QA Steps

- [x] Verify that no errors appear in the JS console

Same as in the **Tests** section

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x]  I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explxanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

android_sender

https://github.com/Expensify/App/assets/23176449/c17c9e34-ca85-4602-8041-68cee1ab197a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/23176449/bf753af6-0113-4643-b033-537ad5e3cdbd

android_web_receiver1

android_web_sender

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/23176449/ef317379-1a1f-430f-96c4-0c2fa3695b60

ios_sender

</details>

<details>
<summary>iOS: mWeb Safari</summary>

ios_web_sender

https://github.com/Expensify/App/assets/23176449/c37be27f-f410-4316-af60-4e31a94b7850

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

web_sender

https://github.com/Expensify/App/assets/23176449/afad9566-916d-4428-9aee-e3a61148171c

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/23176449/cad3000f-e1d8-43bf-afc9-dc10e84ab851

desktop_sender

</details>

### CODE REVIEW

File: src/languages/es.ts
Comment (VickyStash) [inline src/languages/es.ts:677]: I've asked about translation confirmation in the [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1718281196269319)

File: src/components/SettlementButton.tsx
Comment (kubabutkiewicz) [inline src/components/SettlementButton.tsx:277]:
```suggestion
            onSuccessfulKYC={onPress}
```
I think it will work right?

**Comment by jjcoffee** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by jjcoffee** [comment]
@VickyStash Testing well so far! I'm seeing an issue though, where the first invoice sent initially displays with the correct (personal) avatar of User B. However, once the `SendInvoice` request completes it switches to this avatar without any title:

Screenshot from 2024-06-13 16-34-44

User B sees the correct avatars:

image

Refreshing also fixes the issue.

**Comment by VickyStash** [comment]
> Testing well so far! I'm seeing an issue though, where the first invoice sent initially displays with the correct (personal) avatar of User B. However, once the `SendInvoice` request completes it switches to this avatar without any title

@jjcoffee It's a known issue caused by API returns empty `invoiceReceiver` for the invoice room report in the `SendInvoice` response. Since in this PR uses room `invoiceReceiver` data to show avatars it's so noticeable.

cc @cristipaval Have you had a chance to take a look at it?

**Comment by jjcoffee** [comment]
@VickyStash Gotcha, thanks! One other thing, should it work to send a new invoice within the workspace-to-individual chat that's initially created after you send the first invoice? It doesn't seem to come through for User B.

1. As User A, go to FAB -> Send invoice to User B
2. As User B pay the invoice
3. As User A, switch back to the chat created in step 1
4. Use the inline `+` in the composer to send a new invoice
5. Check User B's chat

**Comment by jjcoffee** [comment]
`BUG?:` After sending the initial invoice from `Workspace 1`, invoices sent (using FAB -> Send Invoice) from any other workspace (e.g. `Workspace 2`) don't come through to User B.

**Comment by VickyStash** [comment]
> @VickyStash Gotcha, thanks! One other thing, should it work to send a new invoice within the workspace-to-individual chat that's initially created after you send the first invoice? It doesn't seem to come through for User B.
> 
> 1. As User A, go to FAB -> Send invoice to User B
> 2. As User B pay the invoice
> 3. As User A, switch back to the chat created in step 1
> 4. Use the inline `+` in the composer to send a new invoice
> 5. Check User B's chat

I think I've tested this use case and it worked for me, could you provide a video please 🙏

**Comment by jjcoffee** [comment]
> I think I've tested this use case and it worked for me, could you provide a video please 🙏

@VickyStash Sure! It's most likely a pusher issue as I don't see anything coming through. (The invoice that shows for User B is an older request)

https://github.com/Expensify/App/assets/27287420/7dba164b-8f92-4c70-b77f-fa031c8f18f9

**Comment by VickyStash** [comment]
> `BUG?:` After sending the initial invoice from `Workspace 1`, invoices sent (using FAB -> Send Invoice) from any other workspace (e.g. `Workspace 2`) don't come through to User B.

@jjcoffee It looks like it works for me, did I miss something?

https://github.com/Expensify/App/assets/23176449/04a4c0d2-05ff-4085-9969-0a523b7a4493

**Comment by jjcoffee** [comment]
> It looks like it works for me, did I miss something?

@VickyStash Hmm, interesting! What if you try with User B as a new account that User A has never chatted with before?

**Comment by jjcoffee** [comment]
@VickyStash I guess this is related to the empty `invoiceReceiver` from [here](https://github.com/Expensify/App/pull/43036#issuecomment-2165889167)? User A briefly sees their own workspace name as the payer in the `... paid:` text.

https://github.com/Expensify/App/assets/27287420/ca6c8323-a740-4197-bdcc-fd507ceca305

**Comment by jjcoffee** [comment]
@VickyStash Have we considered this scenario?

1. Two invoices sent by User A to user B
2. User B pays one invoice via business
3. User B pays the other one as personal

Currently after (2) we switch to a B2B chat, but it's still possible to pay as personal (3). Doing (3) switches User B back to person-to-workspace chat, but User A remains on a B2B chat.


https://github.com/Expensify/App/assets/27287420/5cf246ba-240b-4fb2-9dad-8ef2ac8adec7

**Comment by VickyStash** [comment]
> @VickyStash Have we considered this scenario?
> 
> 1. Two invoices sent by User A to user B
> 2. User B pays one invoice via business
> 3. User B pays the other one as personal
> 
> Currently after (2) we switch to a B2B chat, but it's still possible to pay as personal (3). Doing (3) switches User B back to person-to-workspace chat, but User A remains on a B2B chat.
> 
>  desktop-app-mixed-personal-business-2024-06-14_11.41.58.mp4

I was able to reproduce it, the reason is in the optimistic data on the FE.
But before I implement the fix I just want to confirm:
@cristipaval  Should user be able to Pay as individual in B2B room?  Right now it's possible, and the room continue to be B2B.

**Comment by jjcoffee** [comment]
@VickyStash Another thing I'm wondering is should there be a new workspace chat created each time a new invoice is sent (personally to User B)? It seems like ideally if there's an existing B2B chat, the invoice should be added to it to avoid cluttering the LHN like this:

image

**Comment by cristipaval** [comment]
> @cristipaval Should user be able to Pay as individual in B2B room? Right now it's possible, and the room continue to be B2B.

This is a great question! The backend doesn't support this `Pay as Individual` for an invoice sent to a B2B room. Let's fix this on the front end for now and discuss on Slack if we want to enable this feature in separate PRs.

**Comment by VickyStash** [comment]
> I guess this is related to the empty invoiceReceiver from https://github.com/Expensify/App/pull/43036#issuecomment-2165889167? User A briefly sees their own workspace name as the payer in the ... paid: text.

@jjcoffee I've checked - it's truly related `invoiceReceiver: null` issue

**Comment by cristipaval** [comment]
> @VickyStash Another thing I'm wondering is should there be a new workspace chat created each time a new invoice is sent (personally to User B)? It seems like ideally if there's an existing B2B chat, the invoice should be added to it to avoid cluttering the LHN like this:
> 
> image

@jjcoffee I think you're referring to the invoice rooms (they are similar but different than workspace chats 😅 ).

I had an internal discussion with @VickyStash and we think it makes sense to implement the Pay feature in a way that there are at most one b2b and one b2c invoice rooms between the userB as an individual and their primary workspace.

**Comment by VickyStash** [comment]
Just to summarise issues as for now (I'll update this comment as we move forward):

1. **[API]** API returns empty invoiceReceiver for the invoice room report in the SendInvoice response, which results in:
1.1 invoice sent initially displays with the correct (personal) avatar of User B - [comment](https://github.com/Expensify/App/pull/43036#issuecomment-2165877709)
1.2 User A briefly sees their own workspace name as the payer in the ... paid: text. - [comment](https://github.com/Expensify/App/pull/43036#issuecomment-2167647363)
2. **[Need to reproduce]** Sending a new invoice within the workspace-to-individual chat that's initially created after you send the first invoice - [comment](https://github.com/Expensify/App/pull/43036#issuecomment-2165920415)
3. **[Need to reproduce]** After sending the initial invoice from Workspace 1, invoices sent (using FAB -> Send Invoice) from any other workspace (e.g. Workspace 2) don't come through to User B. - [comment](https://github.com/Expensify/App/pull/43036#issuecomment-2165932994)
4. **[DONE]** Bug with paying as individual in B2B room - [comment](https://github.com/Expensify/App/pull/43036#issuecomment-2167664755)
5. **[Additional updates are needed]** B2B invoice rooms duplicates - [comment](https://github.com/Expensify/App/pull/43036)

**Comment by VickyStash** [comment]
The PR is on hold till the API updates to support at most one b2b and one b2c invoice room between the sender and the receiver are implemented. ([comment](https://github.com/Expensify/App/pull/43036#issuecomment-2168118722))

**Comment by VickyStash** [comment]
Note: I'll be OOO on Monday (24.06) 🌴

**Comment by VickyStash** [comment]
@jjcoffee Could you please take another look at the PR?

**Comment by jjcoffee** [comment]
`BUG:` Avatars and name for the room don't update for User A after User B pays as business.

image

After refreshing this is resolved:

image

**Comment by jjcoffee** [comment]
This might be more one for a followup. For cases where we send an invoice to user B again (directly to their personal email), we probably want to redirect to the invoice room if one already exists.

1. As User A, go to FAB -> Send invoice to User B
2. As User B pay the invoice
3. As User A, repeat step 1 making sure to use User B's email, not the newly created invoice room
4. As User B pay the invoice
5. Observe as User A that the room becomes empty (as the invoice is moved to the invoice room)

https://github.com/Expensify/App/assets/27287420/80e883ef-8d38-46a2-bee2-1e7525c423e5

**Comment by jjcoffee** [comment]
Just to cover all bases, if User B has no workspace set up yet should they actually be able to pay as business? Currently this will create a B2B room with `User B's Expenses` as the title. cc @cristipaval

**Comment by VickyStash** [comment]
> Just to cover all bases, if User B has no workspace set up yet should they actually be able to pay as business? Currently this will create a B2B room with `User B's Expenses` as the title. cc @cristipaval

@jjcoffee The user should be able to pay as a business only if he is an admin of his primary workspace for now. There is a [condition](https://github.com/Expensify/App/pull/43036/files#diff-475fc90a9838cf6b734f4c860eb57036c3fe04ac7b4c429d1b4e361109aaa6f4R222) in the app, but it doesn't work for you, right?
I can double-check in 1-2 hours.

**Comment by jjcoffee** [comment]
`BUG:` I'm still seeing the issue with the avatar and name (shows as User A's workspace) initially displaying incorrectly after User B pays. I'm seeing this when User B is a completely new account.

https://github.com/Expensify/App/assets/27287420/268c6d84-795a-4a28-b8de-c5b06b055ee0

**Review by jjcoffee — CHANGES_REQUESTED** [review]
A few issues still outstanding.

**Comment by VickyStash** [comment]
> A few issues still outstanding.

Thank you! I'll look through them tomorrow.

**Comment by VickyStash** [comment]
> Just to cover all bases, if User B has no workspace set up yet should they actually be able to pay as business? Currently this will create a B2B room with `User B's Expenses` as the title. cc @cristipaval

@jjcoffee I've re-checked and I don't see option to `Pay as business` in the account without workspaces. Is there anything special about your account?
image

**Comment by VickyStash** [comment]
> `BUG:` If the workspace that User A sends the invoice from has a custom avatar, this does not reflect in User B's chat.

Same issue is reproduced on the main branch, so it would be better to fix it separately

**Comment by VickyStash** [comment]
**Issues, that should be fixed by API updates:**

- `BUG`: Avatars and name for the room don't update for User A after User B pays as business. This happens when the invoice room doesn't already exist. ([link](https://github.com/Expensify/App/pull/43036#issuecomment-2194536060))
- `BUG`: I'm still seeing the issue with the avatar and name (shows as User A's workspace) initially displaying incorrectly after User B pays. I'm seeing this when User B is a completely new account. ([link](https://github.com/Expensify/App/pull/43036#issuecomment-2194655580))

**Bug exists on main, should be tracked in a separate ticket:**

- `BUG`: If the workspace that User A sends the invoice from has a custom avatar, this does not reflect in User B's chat. ([link](https://github.com/Expensify/App/pull/43036#issuecomment-2194557431))

**Let's leave it to a follow up:**

- This might be more one for a followup. For cases where we send an invoice to user B again (directly to their personal email), we probably want to redirect to the invoice room if one already exists. ([link](https://github.com/Expensify/App/pull/43036#issuecomment-2194655580))

**Fixed in the app:**

- User is able to Pay as business without workspaces set up ([link](https://github.com/Expensify/App/pull/43036#issuecomment-2194641966))

@jjcoffee @cristipaval 

**Note:** I'm going to be OOO Jul 1- Jul 7 🌴

**Comment by cristipaval** [comment]
Thanks for the summary @VickyStash!

@jjcoffee, I think you can approve the PR, and I can test it when the backend fixes get into production. If nothing else occurs, I'll merge this PR and @VickyStash will create a follow-up issue to fix the redirection when the invoice is moved to an existing b2b invoice room. This way you can go ooo and don't worry about this PR.

**Review by jjcoffee — APPROVED** [review]
LGTM!

**Review by cristipaval — APPROVED** [review]

**Comment by Julesssss** [comment]
> @mountinyThe QA team is only seeing the "Pay as individual" option. This was reverted in [this PR](https://github.com/Expensify/App/pull/44891). Are we good to check it off?

@cristipaval could you please verify this before closing the issue? In the meantime I am checking this off our checklist as it isn't a blocker.

**Comment by cristipaval** [comment]
> > @mountinyThe QA team is only seeing the "Pay as individual" option. This was reverted in [this PR](https://github.com/Expensify/App/pull/44891). Are we good to check it off?
> 
> @cristipaval could you please verify this before closing the issue? In the meantime I am checking this off our checklist as it isn't a blocker.

This PR was reverted, @Julesssss

**Comment by mountiny** [comment]
This is the PR that should not have been added to the checklist, it was reverted

**Pull request merged**
